source/core/NumericString.cpp

```diff
@@ -257,83 +257,34 @@ void DefaultFormatCode(Int32 count, StaticTypeAndData arguments[], TempStackCStr
-    buffer->AppendUtf8Str(formatUnEscaped, newLength);
-    buffer->AppendCStr("'\n");
-    delete[] formatUnEscaped;
-}
-
-/**
- * Error reporting function for when the Format String is invalid
- * This will clear the current buffer string and return on what format
- * specifier it failed on.
- * */
-void ErrFormatInvalidFormatString(SubString* format, Int32 count, StaticTypeAndData arguments[], StringRef buffer, char formatCode) {
-    char formatSpecifierBuf[3];
-    snprintf(formatSpecifierBuf, 3, "%c", formatCode);
-    buffer->Resize1D(0);
-    buffer->AppendCStr("Error: Invalid format string provided: '%");
-    buffer->AppendCStr(formatSpecifierBuf);
-    buffer->AppendCStr("' in '");
-    IntIndex length = format->Length()*2;
-    Utf8Char* formatUnEscaped = new Utf8Char[length];
-    IntIndex newLength = format->UnEscape(formatUnEscaped, length);
-    buffer->AppendUtf8Str(formatUnEscaped, newLength);
-    buffer->AppendCStr("'\n");
-    delete[] formatUnEscaped;
-}
-
-/**
- * Error reporting function for when there is a type mismatch between 
- * arguments and the format specifiers.
- * This will clear the current buffer string and return on what format
- * specifier it failed on.
- * */
-void ErrMismatchedFormatSpecifier(SubString* format, Int32 count, StaticTypeAndData arguments[], StringRef buffer, char formatCode) {
-    char formatSpecifierBuf[3];
-    snprintf(formatSpecifierBuf, 3, "%c", formatCode);
-    buffer->Resize1D(0);
-    buffer->AppendCStr("Error: Invalid format specifier type mismatch: '%");
-    buffer->AppendCStr(formatSpecifierBuf);
-    buffer->AppendCStr("' in '");
-    IntIndex length = format->Length()*2;
-    Utf8Char* formatUnEscaped = new Utf8Char[length];
-    IntIndex newLength = format->UnEscape(formatUnEscaped, length);
-    buffer->AppendUtf8Str(formatUnEscaped, newLength);
-    buffer->AppendCStr("'\n");
-    delete[] formatUnEscaped;
+// Format Into String/Scan From String (StringFormat/StringCan) error codes
+enum { kFormatArgErr=1, kFormatTypeMismatch = 81, kFormatCodeUnknown, kFormatTooManyFormatSpecs, kFormatTooFewFormatSpecs, kFormatScanFailed };
+
+static Boolean SetFormatError(Int32 errCode, Int32 argNum, char formatCode, ErrorCluster *errPtr) {
+    if (errPtr) {
+        char argBuf[8];
+        snprintf(argBuf, sizeof(argBuf), "%d", argNum+1);
+        errPtr->SetError(true, errCode, errCode == kFormatScanFailed ? "Scan From String" : "Format Into String");
+        if (errCode != kFormatTooManyFormatSpecs && errCode != kFormatTooFewFormatSpecs) {
+            errPtr->source->AppendCStr(" (arg ");
+            errPtr->source->AppendCStr(argBuf);
+            errPtr->source->AppendCStr(")");
+        }
+        errPtr->source->AppendCStr(" in ");
+        AppendCallChainString(errPtr->source);
+        if (formatCode && (errCode == kFormatCodeUnknown || errCode == kFormatTypeMismatch)) {
+            errPtr->source->AppendCStr("\n<APPEND>\n");
```

Comment:
Sure, I'll add one, but it'll really just be a helper for the APPEND tag, because having it take a string to add after the append would only cause us to always have to make an extra copy of the string if it needed any formatting.